charts/prometheus/templates/service.yaml

```
@@ -29,6 +29,9 @@ spec:
   {{- range $cidr := .Values.server.service.loadBalancerSourceRanges }}
     - {{ $cidr }}
   {{- end }}
+{{- end }}
+{{- if ne .Values.server.service.type "ClusterIP" }}
+  externalTrafficPolicy: {{ .Values.server.service.externalTrafficPolicy }}
```

I'd suggest to apply the configuration only if set by the user:

```suggestion
{{- if .Values.server.service.externalTrafficPolicy }}
  externalTrafficPolicy: {{ .Values.server.service.externalTrafficPolicy }}
```